examples/managed_instance_group/main.tf

```diff
@@ -0,0 +1,130 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+provider "google" {
+  credentials = "${file(var.credentials_path)}"
+  region      = "${var.region}"
+}
+
+module "gce-container" {
+  source = "../../../"
+
+  container = {
+    image = "${var.image}"
+  }
+}
+
+resource "tls_private_key" "gce-keypair" {
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+resource "local_file" "gce-keypair-pk" {
+  content  = "${tls_private_key.gce-keypair.private_key_pem}"
+  filename = "${path.module}/ssh/key"
+}
+
+module "mig" {
+  source             = "GoogleCloudPlatform/managed-instance-group/google"
+  version            = "1.1.14"
+  project            = "${var.project_id}"
+  region             = "${var.region}"
+  zone               = "${var.zone}"
+  name               = "${var.mig_name}"
+  machine_type       = "${var.machine_type}"
+  compute_image      = "${module.gce-container.source_image}"
+  size               = "${var.mig_instance_count}"
+  service_port       = "${var.image_port}"
+  service_port_name  = "http"
+  http_health_check  = "${var.enable_http_health_check ? true : false}"
+  subnetwork         = "${var.subnetwork}"
+  subnetwork_project = "${var.subnetwork_project}"
+  ssh_source_ranges  = ["0.0.0.0/0"]
+  target_tags        = ["container-vm-test-mig"]
+
+  metadata = {
+    "gce-container-declaration" = "${module.gce-container.metadata_value}"
+    sshKeys                     = "${var.gce_ssh_user}:${tls_private_key.gce-keypair.public_key_openssh}"
+  }
+
+  instance_labels = {
+    "container-vm" = "${module.gce-container.vm_container_label}"
+  }
+
+  service_account_scopes = [
+    "https://www.googleapis.com/auth/devstorage.read_only",
+    "https://www.googleapis.com/auth/logging.write",
+    "https://www.googleapis.com/auth/monitoring.write",
+    "https://www.googleapis.com/auth/servicecontrol",
+    "https://www.googleapis.com/auth/service.management.readonly",
+    "https://www.googleapis.com/auth/trace.append",
+    "https://www.googleapis.com/auth/cloud-platform",
+  ]
+
+  wait_for_instances = true
+}
+
+module "http-lb" {
+  source            = "github.com/GoogleCloudPlatform/terraform-google-lb-http"
+  project           = "${var.project_id}"
+  name              = "${var.mig_name}-lb"
+  firewall_networks = ["${var.subnetwork}"]
+  target_tags       = ["${module.mig.target_tags}"]
+
+  backends = {
+    "0" = [
+      {
+        group = "${module.mig.instance_group}"
+      },
+    ]
+  }
+
+  backend_params = [
+    "/,http,${var.image_port},30",
+  ]
+}
+
+locals {
+  google_load_balancer_ip_ranges = ["130.211.0.0/22", "35.191.0.0/16"]
+}
+
+resource "google_compute_firewall" "lb-to-instances" {
+  name    = "container-mig-firewall-lb-to-instances"
+  project = "${var.project_id}"
+  network = "${var.subnetwork}"
```

Comment:
When we reuse `var.subnetwork` for both the GCE network and subnetwork, we make the assumption about the network naming that won't hold. I ran into this because I use the project-factory to stand up fixture projects, which deletes the default network by default. To re-add networks I'm using the network module which doesn't inherently define a `default` network and `default` subnets.

Don't worry about changing this; I'm going to just match this convention - this comment is more of a heads up about this behavior.